### ISSUE TITLE
Handle `httpx.PoolTimeout` on static analysis upload

### ISSUE THREAD

**Issue by giovanni-guidini** [codecov/engineering-team#652]
When uploading the individual files to GCS sometimes we see `httpx.PoolTimeout`.
This doesn't happen on every repo though. It might be related to https://github.com/encode/httpx/issues/1171

In any case it would be better to handle it more gracefully, as this impacts the Static Analysis upload process directly.

One suggestion was to try and upload the files in batches to GCS. Would that be possible?

**Comment by giovanni-guidini** [comment]
@scott-codecov noted that it's not possible to batch upload files. See [Google Cloud Docs](https://cloud.google.com/python/docs/reference/storage/latest/google.cloud.storage.batch)

**Comment by giovanni-guidini** [comment]
Stacktrace
```
Uploading files  [####################################]  100%
warning - 2023-10-05 19:05:54,993 -- Unable to send single_upload_put --- {"message": "", "exception": "<class 'httpx.PoolTimeout'>", "filepath": "test/vendor/__init__.py", "latest_status_code": null}
warning - 2023-10-05 19:05:54,993 -- Unable to send single_upload_put --- {"message": "", "exception": "<class 'httpx.PoolTimeout'>", "filepath": "scripts/core/cst/__init__.py", "latest_status_code": null}
warning - 2023-10-05 19:05:54,994 -- Unable to send single_upload_put --- {"message": "", "exception": "<class 'httpx.PoolTimeout'>", "filepath": "subscription_server/controllers/__init__.py", "latest_status_code": null}
warning - 2023-10-05 19:05:54,994 -- Unable to send single_upload_put --- {"message": "", "exception": "<class 'httpx.PoolTimeout'>", "filepath": "subscription_service/scripts/__init__.py", "latest_status_code": null}
```

**Comment by giovanni-guidini** [comment]
We tried to fix this releasing [CLI v0.3.8](https://github.com/codecov/codecov-cli/releases/tag/0.3.8), which got rid of the `https.Limits` and went back to the default ones.

Now the error is `httpx.ReadTimeout`
```
warning - 2023-10-12 22:33:28,050 -- Unable to send single_upload_put --- {"message": "", "exception": "<class 'httpx.ReadTimeout'>", "filepath": "", "latest_status_code": null}
warning - 2023-10-12 22:33:28,054 -- Unable to send single_upload_put --- {"message": "", "exception": "<class 'httpx.ReadTimeout'>", "filepath": "", "latest_status_code": null}
warning - 2023-10-12 22:33:28,058 -- Unable to send single_upload_put --- {"message": "", "exception": "<class 'httpx.ReadTimeout'>", "filepath": "", "latest_status_code": null}
```
